src/nxdoc/nuxeo-server/administration/variables-and-configuration/variables-and-configuration-pattern.md

```diff
@@ -0,0 +1,300 @@
+
+1. The map is initialized with properties held by file at `NUXEO_HOME/templates/nuxeo.defaults`.
+1. If an environment variable `NUXEO_ENVIRONMENT` is set, for instance to value `prod`, the map is then filled with properties held by file at `NUXEO_HOME/templates/nuxeo.prod`.
+1. The map is then filled with System properties via `System#getProperties`.
+1. The map is then filled with `nuxeo.conf` variable properties.
+1. Then templates are resolved thanks to the property `nuxeo.templates` (*).
+This resolution supports “included templates” resolution thanks to the `nuxeo.template.includes` property resolution. Templates are ordered so that they follow the declaration order, and so that included templates are taken into account before the template including them.
+1. If an environment variable `NUXEO_PROFILES` is set, the value is added to the list of templates previously resolved. Multiple profiles can be defined, separated by a comma.
+1. For each template, following the order of the list of template names previously resolved, the map is then filled by properties held by file at `TEMPLATE_DIRECTORY/nuxeo.defaults`, and if an environment variable `NUXEO_ENVIRONMENT` is set, for instance to value `prod`, the map is then filled with properties held by file at `TEMPLATE_DIRECTORY/nuxeo.prod`.
+Note that `TEMPLATE_DIRECTORY` can either stand for the template name (in which case the sub-directory with this name will be resolved inside the `NUXEO_HOME/templates/` directory), or an absolute directory path.
+
+These steps are covered by environment variables expansion explained below.
+
+### Pre-processing Environment Variables Expansion
+
+During preprocessing, properties pushed to the main map can be [expanded](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java#L698) when they are declared with the pattern `${env:VARIABLE_NAME}` or `${env:VARIABLE_NAME:defaultValue}`, for `nuxeo.conf` templates and for properties held by templates.
+
+Values are replaced according to environment variables: for the given property name, the value returned by System#getEnv will be used, and will default to the `defaultValue` if specified, or to an empty string.
+
+Although all resolutions can follow this pattern, this is specifically useful to declare additional Nuxeo configuration templates:
+
+```
+nuxeo.templates=default,common,testinclude,${env:TEST_ENV:testenv}
+```
+
+Note that in the `nuxeo.conf` file, only properties referring to templates are covered by this expansion: all properties might need to be covered too, to be consistent with template properties resolution.
```

Comment:
This is more a potential "TODO" personal comment, maybe it should be rephrased for a public doc page.